task4/src/components/movie/movie.js

```diff
@@ -1,21 +1,44 @@
 import React from 'react';
 import './movie.scss';
 import PropTypes from 'prop-types';
+import EditMovie from '../editMovie/editMovie';
 
-function Movie(props) {
-  const {poster = '', title = '', genre = '', year = ''} = props.movie;
-  return (
-    <div className="movie">
-      <img src={poster}/>
-      <div className="movie-info">
-        <h3>
-          {title}
-          <span>{genre}</span>
-        </h3>
-        <span className="year">{year}</span>
+class Movie extends React.Component{
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      movie: props.movie,
+      showModal: false,
+    };
+  }
+
+  toggleModal(value) {
+    console.log(value || !this.state.showModal);
+    this.setState({
+      showModal: value || !this.state.showModal
+    });
+  }
+
+  render() {
+    return (
+      <div className="movie"
+           onClick={this.toggleModal.bind(this, !this.state.showModal)}>
+        <img src={this.props.movie.poster}/>
+        <div className="movie-info">
+          <h3>
+            {this.props.movie.title}
+            <span>{this.props.movie.genre}</span>
+          </h3>
+          <span className="year">{this.props.movie.year}</span>
+        </div>
+        <EditMovie isOpen={this.state.showModal}
+                   editItem={this.props.movie}
+                   getsClosed={this.toggleModal.bind(this, false)}/>
```

Comment:
to bind this function you can do it in the constructor

```
constructor(props){
    super(props);
    this.toggleModal = this.toggleModal.bind(this);
}
```

but normally react developers do not like to bind functions so we use arrow functions in class fields

```
  toggleModal = (value) => {
    console.log(value || !this.state.showModal);
    this.setState({
      showModal: value || !this.state.showModal
    });
  }
```